lib/PuppeteerSharp.Tests/PageTests/EvaluateTests.cs

```diff
@@ -241,5 +242,38 @@ public async Task ShouldThrowANiceErrorAfterANavigation()
             });
             Assert.Contains("navigation", ex.Message);
         }
+
+        [Fact]
+        public async Task ShouldWorkWithoutGenerics()
+        {
+            var objectEmpty = await Page.EvaluateExpressionAsync("{}");
+            var arrayEmpty = await Page.EvaluateExpressionAsync("[]");
+            var stringEmpty = await Page.EvaluateExpressionAsync("''");
```

Comment:
Can we support returning empty arrays and empty strings?

Reply:
Yes, it would require a change to that new method, I am kinda on the fence as to which is the expected behavior of a user ?

Reply:
There is some expectation in the unit tests already I believe

Reply:
I think it would be more clear for the user if `Page.EvaluateExpressionAsync` is a `void/Task` method. If you call that you want to execute you don't want to return data.
If you want to return data use `Page.EvaluateExpressionAsync<T>` instead.

Thoughts @Meir017 ?

Reply:
I think we discussed this in the original PR, if we wanted to add these overloads we would need to differentiate the names
In the interest of patching this bug, we should probably tackle that separately.

Reply:
Fair enough, so let's avoid treating things that are not null as null: `{}`, `[]`, `''` should not return null.

Reply:
I know that there are unit tests that expect empty object `{}` to be null, do you want them changing ?

Reply:
Which one?

Reply:
I was wrong, there are 3x tests that expect empty array to be null

`InputTests.ShouldClickOnCheckboxInputAndToggle`
`InputTests.ShouldClickOnCheckboxLabelAndToggle`
`EvaluateTests.ShouldWorkWithoutGenerics`

Reply:
result.check is null https://github.com/kblok/puppeteer-sharp/blob/master/lib/PuppeteerSharp.TestServer/wwwroot/input/checkbox.html#L11 

EvaluateTests.ShouldWorkWithoutGenerics is yours :joy:

Reply:
Ok, I've made the change, actually simplifies the change and removes the need for the method.

It does however mean that this becomes a breaking change we should mention in change notes.

Reply:
Great Job @tommymonk !